src/main/webapp/script.js

```diff
@@ -71,15 +68,14 @@ function postData(url, data) {
         body: data
       }
   )
-      .then((response) => {
-        if (!response.ok) {
-          throw new Error('Network error detected');
-        }
-        return response.json();
-      })
+      .then((response) => getJsonIfOk(response))
       .catch((error) => handleFetchError(error));
 }
 
-function handleFetchError(exception) {
```

Comment:
This function shouldn't be deleted, it is still used.

Reply:
And, by the way, it should either re-throw the exception or return `Promise.reject(exception)`. Otherwise it swallows the exception and code calling `postData()` would have then-handlers executed even in case of network failure.

Reply:
I know about this, but there was a similar question in the portfolio development, and I think eventually we decided that in case it breaks it is not a problem that the loader function fails.